pkgs/tools/inputmethods/fcitx5/fcitx5-lua.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, fetchFromGitHub
+, cmake
+, extra-cmake-modules
+, fcitx5
+, lua5_3
+, gettext
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fcitx5-lua";
+  version = "5.0.0";
+
+  src = fetchFromGitHub {
+    repo = "fcitx";
+    owner = "fcitx5-lua";
+    rev = "${version}";
+    sha256 = "sha256-ABQS2DaOpkjyMla9V3CU87PlreRnB4KXNan2PjJ7vWU=";
+  };
+
+  # More hacks
+  preInstall = ''
+    sed -i "s#${fcitx5}#$out#g" src/addonloader/cmake_install.cmake
```

Comment:
set cmake flag `-DFEM_INCLUDE_INSTALL_DIR` and `-DFEM_LIB_INSTALL_DIR` can avoid hack generated files. You also need to substitute the pattern `${FCITX_INSTALL_PKGDATADIR}` in `CMakeLists.txt`.

Reply:
I'm getting this error:
```
CMake Error: Parse error in command line argument: -DFEM_INCLUDE_INSTALL_DIR
```
Are you sure that that is an actual CMake flag?

Reply:
How do you add these flags?

I think you should use 

```
cmakeFlags = [ "-DFEM_INCLUDE_INSTALL_DIR=$out/include" "-DFEM_LIB_INSTALL_DIR=$out/lib" ]
```

Reply:
The flags is defined here https://github.com/fcitx/fcitx5/blob/e24005720db378534fb92129e7f168d3186fe3fc/src/lib/fcitx-utils/Fcitx5Macros.cmake#L59

used by all fcitx5 plugin(if they use cmake and macro `fcitx5_export_module`)

Reply:
I was being dumb XD. Fixed now.